### PR TITLE
[ruby 2.4] Implemented Integer#digits

### PR DESCRIPTION
**Pull request by herwinw**
It passes all the following tests of TestInteger of MRI:

* test_digits
* test_digits_for_negative_numbers
* test_digits_for_invalid_base_numbers
* test_digits_for_non_integral_base_numbers
* test_digits_for_non_numeric_base_argument

### CODE REVIEW

File: core/src/main/java/org/jruby/RubyInteger.java [hunk 1]
```diff
@@ -515,6 +515,50 @@ public IRubyObject gcdlcm(ThreadContext context, IRubyObject other) {
...
     }
 
+    @JRubyMethod(name = "digits")
+    public RubyArray digits(ThreadContext context) {
+        return digits(context, RubyFixnum.newFixnum(context.getRuntime(), 10));
+    }
+
+    @JRubyMethod(name = "digits")
+    public RubyArray digits(ThreadContext context, IRubyObject base) {
+        long self = ((RubyFixnum)this).getLongValue();
```
Comment (enebo) [inline core/src/main/java/org/jruby/RubyInteger.java]: Is this guanteed to be an instance of RubyFixnum?  It can also be a RubyBignum.

**Comment by herwinw** [comment]
I don't expect to have a chance to fix this in the next two weeks. If someone wants to finish this: be my guest. Otherwise, it might have to wait a little bit

**Comment by whwilder** [comment]
I've written a fix for this issue, however I've never contributed to someone else's pull request.  I assume I submit a pull request to @herwinw's repository?

**Comment by kares** [comment]
@whwilder that assumes he will than manage the PR out. and since this is stale for 2 months now :
you could also just pick his commit and address the concerns raised on top of them (in a separate PR)

**Comment by headius** [comment]
Yeah, like @kares said, it may be best to just create a new PR based on this one. Github doesn't have a great way for two people in two repos to collaborate on a single PR.

**Comment by herwinw** [comment]
This is continued in #4490, no need to keep this PR opened

**Pull request closed without merging**